models/issue_label.go

```diff
@@ -274,13 +363,23 @@ func GetLabelIDsInRepoByNames(repoID int64, labelNames []string) ([]int64, error
 		Find(&labelIDs)
 }
 
-// GetLabelIDsInReposByNames returns a list of labelIDs by names in one of the given
-// repositories.
-// it silently ignores label names that do not belong to the repository.
+// GetLabelIDsInReposByNames returns a list of labelIDs that are availabble to repositories
+// based on name. This will check for organization labels the repo has access to as well
+// and include them in the results
 func GetLabelIDsInReposByNames(repoIDs []int64, labelNames []string) ([]int64, error) {
+
+	ownerIDs := make([]int64, 0, len(repoIDs))
+	x.Table("repository").
+		In("repository.id", repoIDs).
+		Cols("owner_id").
+		Distinct("owner_id").
+		Join("INNER", "org_user", "`org_user`.org_id = `repository`.owner_id").
+		Find(&ownerIDs)
+
 	labelIDs := make([]int64, 0, len(labelNames))
 	return labelIDs, x.Table("label").
 		In("repo_id", repoIDs).
+		Or(builder.In("org_id", ownerIDs)).
```

Comment:
The results from the above query should be included here as a subquery instead. I think there's a couple of examples of how to do that in other parts of `models`.

Having said that... where is this function used? I don't believe it makes sense to retrieve a list of IDs of labels that are not matched to any particular context. 🤔

Reply:
This is used in the IssuesSearch api that searches any issues a user can view:

 https://try.gitea.io/api/swagger#/issue/issueSearchIssues

I've fixed added tests for this api with labels as well because it was actually broken in a previous commit, see PR description.

And see routers/api/v1/repo/issue.go in this PR

Will look into using a sub query!

Reply:
I've updated to use a subquery